docs/source/graph_preprocessor.rst

```diff
@@ -0,0 +1,41 @@
+The date of the transactions as well as the amount transferred between the accounts may be provided as edge features.
+However, the GraphFeaturePreprocessor can generate a much richer set of edge features by analyzing the graph topology.   
+For instance, the GraphFeaturePreprocessor can extract the number of cycles each transaction is a part of in the graph.
+Such additional features can significantly increase the predictive power of machine learning models in practical settings. 
+
+.. image:: graph-feature-extraction.png
+  :width: 800
+  :align: center
+  :alt: Graph feature extraction 
+
+The current version of the GraphFeaturePreprocessor is specialised to operate on temporal graphs. 
+More specifically, the graph edges must have timestamps. Furthermore, each edge must have a unique id. 
+The edge list must be provided as a two dimensional numpy array of floating point numbers.
+The first column must be `Edge id`, the second `Source vertex id`, the third `Target vertex id`, and the fourth `Timestamp`.     
+Additional columns can be used to store additional edge features.
+
++-------------------+-------------------+-------------------+-------------------+------------------+-------------------+
+| Edge id           | Source vertex id  | Target vertex id  | Timestamp         |  Other Feature   | Other Feature     |
++===================+===================+===================+===================+==================+===================+
+| 1000              | 1                 | 2                 | 10                |                  |                   |
++-------------------+-------------------+-------------------+-------------------+------------------+-------------------+
+| 3000              | 3                 | 4                 | 20                |                  |                   |
++-------------------+-------------------+-------------------+-------------------+------------------+-------------------+
+| 2000              | 1                 | 4                 | 20                |                  |                   |
++-------------------+-------------------+-------------------+-------------------+------------------+-------------------+
+| 4000              | 3                 | 2                 | 30                |                  |                   |
++-------------------+-------------------+-------------------+-------------------+------------------+-------------------+
+
+The edge list must be sorted in the ascending order of the timestamps for best performance. 
+An in-memory directed multi-graph representation will be created internally when an edge list is provided as input to the `fit` function.
+The `transform` function, on the other hand, receives an edge list as input, inserts these edges into the in-memory graph,
+and extracts new features from the graph, which are appended to the input edge list as additional columns to form the output of this function.   
+
```

Comment:
would it make sense to add also a code snippet that shows the sequence of the steps described above?  something like:

```
from snapml import GraphFeaturePreprocessor

# Instatiate a graph feature preprocessor object
gp = GraphFeaturePreprocessor()

# define the configuration of the graph preprocessor
params = {
    "num_threads": 12,  # number of software threads to be used
    "time_window": 10 * 3600,  # time window used if no pattern was specified
    "scatter-gather": True,
    "scatter-gather_tw": 120 * 3600,
    "scatter-gather_bins": [ 2, 3 ],
}

# set the parameters of the greaph feature preprocessor
gp.set_params(params)

# load old transactions
many_old_transactions = np.loadtxt(old_transactions_file_path)

# Create the graph out of the transactions loaded above
gp.fit(many_old_transactions)

# load an unknown transaction, here X_test comprises one transaction
new_transaction = np.loadtxt(new_transaction_file_path)
enriched_new_transaction = gp.transform(new_transaction)
```

Reply:
I did not see code snippets in other rest files. Don't we use examples for that purpose?

Reply:
True, the examples repository is the main place where we store the code examples. However, if it will take long to have the example ready (e.g., due to datasets not being publicly available), then we could (temporarily) provide at least this code snippet in the documentation. What do you think?